### PR TITLE
[Android/iOS] Handle dynamically switching between item layouts in CollectionView

### LINKED ISSUE #5354: [CollectionView] Updating the ItemsLayout type should refresh the layout
More a work task than a bug due to the ongoing development of `CollectionView` (but we'll just categorize it as a bug for now), it should update its layout when switching the `ItemsLayout` types from a `ListItemsLayout` to `GridItemsLayout`. A good example of where this would be used is in a UI where the user is allowed to toggle between those layouts, such as in a retail app. 

This is supposedly semi-functional on iOS but wasn't personally tried. Android doesn't update the layout.

### Steps to Reproduce

1. Use a CollectionView with one of the two layout types
2. Switch to the other layout type in app (such as via a button tap)

### Expected Behavior
The CollectionView updates its layout to the new type.

### Actual Behavior
The CollectionView does not update its layout until leaving and reentering the page (such as with tabs).

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change ###

If you switch between linear and grid layouts, the view should be updated now. Please note that this does not implement UWP.

### Issues Resolved ### 

- fixes #5354

### Testing Procedure ###
Run the attached test and switch between item layouts. Verify that the spacing was not lost.

### PR Checklist ###

- [x] Targets the correct branch
- [x] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/CollectionView/ItemsViewRenderer.cs [hunk 1]
```diff
@@ -591,6 +591,15 @@ protected virtual void ScrollTo(ScrollToRequestEventArgs args)
 			}
 		}
 
+		protected virtual void UpdateLayoutManager()
+		{
+			ItemsLayout = GetItemsLayout();
+			SetLayoutManager(SelectLayoutManager(ItemsLayout));
+
+			UpdateFlowDirection();
+			UpdateItemSpacing();
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/CollectionView/ItemsViewRenderer.cs]: Not sure if we need to update anything else here. Any thoughts?

**Comment by rmarinho** [comment]
Hey @adrianknight89 can you add a UITest? 

Thanks

**Comment by adrianknight89** [comment]
I'll see what I can do later tonight. Thanks.

**Comment by adrianknight89** [comment]
@rmarinho Added a UI test.

**Comment by rmarinho** [comment]
@adrianknight89 I think the merged had some error, the build is failing, can you check again? thanks

**Comment by adrianknight89** [comment]
@rmarinho I rebased. I don't see an issue when I locally build, but I ran `Xamarin.Forms.Controls.Issues.Shared.projitems` through an XML validator and got the same issue as that of build. I'll look at this later today.

In the meantime, feel free to run build again. I don't see a problem with the changes in this PR at first glance.

**Comment by adrianknight89** [comment]
@rmarinho Is it building now? I don't believe those Windows errors are related.

**Pull request merged**